src/Nox.Lib/Presentation/Api/Swagger/ApiRouteMappingDocumentFilter.cs

```diff
@@ -78,39 +75,45 @@ public void Apply(OpenApiDocument openApiDocument, DocumentFilterContext context
                 operation.Parameters.Add(p);
             }
 
-            // add request body if it's supplied
-
-            if (route.JsonBodyType is not null 
-                && route.HttpVerb != HttpVerb.Get)
+            if (route.JsonBodyType is null && route.ResponseOutput is null)
             {
-                operation.RequestBody = new OpenApiRequestBody()
+                AddRequestBodyAndResponse(context, route, operation);
+            }
+            else
+            {
+                // add request body if it's supplied
```

Comment:
we could also extract this to a meaningful method making this function easier to read and understand?